src/Features/Core/Portable/Completion/Providers/Snippets/SnippetCompletionItem.cs

```diff
@@ -36,7 +37,7 @@ internal class SnippetCompletionItem
                 properties: props,
                 isComplexTextEdit: true,
                 inlineDescription: inlineDescription,
-                rules: CompletionItemRules.Default)
+                rules: CompletionItemRules.Default.WithMatchPriority(MatchPriority.Deprioritize))
```

Comment:
I think this would work as you intended for regular items, but in general this won't make completion to select unimported items over snippets (unless the filter text is a complete match for unimported item but not for the snippet)